### ISSUE TITLE
Data set samples are not always displayed in file browser 2

### ISSUE THREAD

**Issue by hackdna** [refinery-platform/refinery-platform#1882]
* Specific code commit: release 1.5.7
* Version of the web browser and OS: Chrome 58, Safari 10 and Firefox 54 on OS X
* Environment where the error occurred: AWS

### Steps to reproduce
1. Import a [tabular data set](http://data.cloud.refinery-platform.org.s3.amazonaws.com/data/sample-data/E-MTAB-4119/e-mtab-4119-local.txt) and upload the data files
1. Run a paired end RNA-seq analysis on the samples (may be optional)
1. Open the data set in File Browser 2 (for example: https://test.stemcellcommons.org/data_sets2/d8f61a90-848e-47a6-9fa1-c73bc3fac513/#/files/)

### Observed behavior
* No samples are displayed
* Web browser console errors:
```
TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at angular.js:38
    at angular.js:30441
    at angular.js:17617
    at m.$digest (angular.js:17755)
    at m.$apply (angular.js:18021)
    at angular.js:18322
    at e (angular.js:6035)
    at angular.js:6314
```

### Expected behavior
Data set samples should be displayed and no errors should be logged in browser console

### Notes
Samples are displayed after clicking on the Analyses or Details tabs before returning to the Files tab

**Comment by mccalluc** [comment]
Confirming that it is not necessary to run an analysis to see this behavior.

**Comment by jkmarx** [comment]
Locally, I reproduced the issue with Data Set 13880: Genome-wide maps of binding sites of Nanog-like and Mxtx2 in blastula stage zebrafish embryos. There's an issue with duplicate attribute names, which I believed was handled at one point. UI-Grid does not allow duplicate column names.

**Comment by jkmarx** [comment]
If I remember correctly, the client is not expecting attributes with the same name and type. Two Notes which are characteristics. <img width="428" alt="screen shot 2017-08-13 at 12 18 56 am" src="https://user-images.githubusercontent.com/9956764/29246115-1ce27498-7fbd-11e7-8d29-ea8973b3fa8a.png">

**Comment by jkmarx** [comment]
@ngehlenborg Please see above. How should the UI display attributes (column headers) with the same name and type? Include the internal keys? Affix a number?

**Comment by ngehlenborg** [comment]
Thanks @jkmarx. I need to get input from @mccalluc on the meaning of the `_generic_` fields. If they are duplicates of the specific fields (indicated by the number combination, e.g. `_30_15_`) they should not show up at all.

**Comment by mccalluc** [comment]
Right: For cross data set browsing to work (faceting and sorting) I needed solr fields with consistent names across data sets. First, I apologize for introducing something which messed up your end. Possible ways forward:

- In FB2, ignore all `_generic_` fields?
- In FB2, ignore non-`_generic_` fields? I had asked earlier, and got the sense the the whole `_#_#_` scheme might not be necessary.
- Set up a new index with just the `_generic_` fields? Only `user/files` will use it initially, but we might have other parts of the code start to use it over time.

**Comment by ngehlenborg** [comment]
It sounds like we will have to adjust the node and attribute APIs to deal with this. I am not sure what is easy, ignoring `_generic_` fields or changing everything to just operate on `_generic_` fields. 

There are probably fewer side effects to ignoring `_generic_` fields for the time being until we can get back to this to use a single approach for both interfaces. 

@jkmarx and @mccalluc: what do you prefer? @mccalluc: can you look into the APIs that feed FB2 to see if you can find an easy solution to ignore generic fields?

**Comment by mccalluc** [comment]
Here's my plan:
- update my dev environment
- reproduce the fb2 not loading problem
- in a junk branch, tweak `prepare` method to not include `_generic_` fields, and see if that fixes it. (Should be an easy test)
- If it does, then revert that local change, reindex again, and try to tweaks APIs and Nils suggests.

**Comment by jkmarx** [comment]
If the api handles it, great. The FB2 UI could be adjusted to filter out the generics. The attributes are already being looped through to create column names and check for duplicates.

**Comment by mccalluc** [comment]
Scott has reproduced it, but I have not. (Before I did a clean install, I did see it, and now I don't.) For him, the API endpoint http://192.168.50.50/api/v2/assays/..../files/?limit=100&offset=0 returns duplicate names under `attributes`... but I do not see the same behavior. TODO: We need to figure out what the difference is between his environment and mine. In parallel, I'll look at the code that generates this response.
